### PR TITLE
profile endpoint

### LINKED ISSUE #396: RPC Endpoint für Profile
- sämtliche editierbare Userdaten ändern 
- Auch beachten #392 
- GUI-Sprache

### LINKED ISSUE #397: i18n im Profil speichern
- im Profil Language String bearbeitbar und abrufbar für translation

### PR DESCRIPTION
**Pull request by pmattmann**
Profile Endpoint with language

### CODE REVIEW

File: backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            $data = call_user_func([$this, $method], $user);
+        }
+
+        $json = new HalJsonModel();
+        $json->setPayload(new Entity($data));
+
+        return $json;
+    }
+
+    public function getAction(User $user) {
```
Comment (usu) [inline backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php]: private

File: backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            ]),
+            'username' => $user->getUsername(),
+            'displayName' => $user->getDisplayName(),
+            'mail' => $user->getTrustedMailAddress(),
+            'role' => $user->getRole(),
+            'language' => $user->getLanguage(),
+        ];
+    }
+
+    public function patchAction(User $user) {
```
Comment (usu) [inline backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php]: private

File: backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        UserService $userService
+    ) {
+        $this->authenticationService = $authenticationService;
+        $this->userService = $userService;
+    }
+
+    public function indexAction() {
+        $data = [];
+
+        if ($this->authenticationService->hasIdentity()) {
```
Comment (usu) [inline backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php:33]: else throw 403

**Review by usu — CHANGES_REQUESTED** [review]

File: backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php [hunk 4]
```diff
@@ -28,8 +30,6 @@ public function __construct(
     }
 
     public function indexAction() {
```
Comment (manuelmeister) [inline backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php:32]: TODO: Description einfügen

File: backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php [hunk 5]
```diff
@@ -40,35 +40,46 @@ public function indexAction() {
...
+    private function getAction(User $user) {
         return [
             'self' => Link::factory([
                 'rel' => 'self',
                 'route' => 'e-camp-api.rpc.profile',
             ]),
             'username' => $user->getUsername(),
+            'firstname' => $user->getFirstname(),
+            'surname' => $user->getSurname(),
+            'scoutname' => $user->getScoutname(),
```
Comment (manuelmeister) [inline backend/module/eCampApi/src/eCampApi/V1/Rpc/Profile/ProfileController.php:59]: TODO: Nickname anstatt Scoutname verwenden

File: frontend/src/views/Profile.vue [hunk 6]
```diff
@@ -24,17 +42,35 @@
...
     ApiTextField,
     ContentCard,
     ButtonBack
   },
   computed: {
     profile () {
       return this.api.get().profile()
+    },
+    availableLocales () {
+      return VueI18n.availableLocales.map(function (l) {
```
Comment (pmattmann) [inline frontend/src/views/Profile.vue:61]: Use Arrow-Function

**Pull request merged**